frontend/lit/components/Reference.js

```diff
@@ -9,6 +9,33 @@ import Hero from "shared/utils/Hero";
 
 import ReferenceButton from "./ReferenceButton";
 
+const markKeywords = (text, allSettings) => {
+        const all_tokens = allSettings.set1.keywords
+                .concat(allSettings.set2.keywords)
+                .concat(allSettings.set3.keywords),
+            all_re = new RegExp(all_tokens.join("|"), "gim");
+
+        if (all_tokens.length === 0) {
```

Comment:
found an edge case too if all tokens are null it highlights everything; so i added a few guards to bail early if that's the case...